src/main/java/org/pac4j/vertx/verticle/DemoServerVerticle.java

```diff
@@ -121,7 +121,13 @@ public void start() {
         // Requires authentication endpoint without specific authenticator attached
         addProtectedEndpointWithoutAuthorizer("/protected/index.html", "", router);
 
-        router.get("/index.html").handler(DemoHandlers.indexHandler(config));
+        // Direct basic auth authentication (web services)
+        addProtectedEndpointWithoutAuthorizer("/dba/index.html", "DirectBasicAuthClient", router);
```

Comment:
Also let me examine the vert.x-web source and see if I can manage a clean override of the behaviour. I’m not keen to copy a whole load of code, but if I can override just a small part of it and get the behaviour I want, I will do so :) That rather depends how extensible they’ve already made their handler. It may also be possible that we can trigger a handler that hooks in after their one to remove the user from the session, but that feels very hacky (and potentially lead to very odd errors).

Let me also see if I can manage it through vert.x config by only having the SessionHandler and UserSessionHandler defined on specific routes rather than all routes. That may provide an alternative (the route exclusion approach we discussed briefly earlier).

Cheers

Jez

> On 29 Dec 2015, at 16:53, LELEU Jérôme <notifications@github.com> wrote:
> 
> In src/main/java/org/pac4j/vertx/verticle/DemoServerVerticle.java <https://github.com/pac4j/vertx-pac4j-demo/pull/35#discussion_r48553164>:
> 
> > @@ -121,7 +121,13 @@ public void start() {
> >          // Requires authentication endpoint without specific authenticator attached
> >          addProtectedEndpointWithoutAuthorizer("/protected/index.html", "", router);
> >  
> > -        router.get("/index.html").handler(DemoHandlers.indexHandler(config));
> > +        // Direct basic auth authentication (web services)
> > +        addProtectedEndpointWithoutAuthorizer("/dba/index.html", "DirectBasicAuthClient", router);
> OK. Thanks for the explanations. There is no security issue with saving the user in session even for a web service call, it is just not necessary. Let vertx-web rule on this and pac4j follow the pace.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/pac4j/vertx-pac4j-demo/pull/35/files#r48553164>.
>